### PR TITLE
SW-3950 Use Site Reported Plants Endpoint

### PR DESCRIPTION
**Pull request by charlie-tf**
- create service
- create redux
- replace plantings date range selector with new selector
- delete extraneous calculations

### CODE REVIEW

File: src/redux/features/tracking/trackingSlice.ts [hunk 1]
```diff
@@ -71,3 +72,27 @@ export const sitePopulationSlice = createSlice({
...
 
 export const sitePopulationReducer = sitePopulationSlice.reducer;
+
+// Planting Site Reported Plants Slice
+type SiteReportedPlantsData = {
+  error?: string;
+  site?: PlantingSiteReportedPlants;
+};
+
+const initialSiteReportedPlantsState: SiteReportedPlantsData = {};
```
Comment (karthikbtf) [inline src/redux/features/tracking/trackingSlice.ts:82]: this should ideally be a record of site id to data

File: src/redux/features/tracking/trackingThunks.ts [hunk 2]
```diff
@@ -49,3 +54,17 @@ export const requestPlantingSitesSearchResults = (organizationId: number) => {
     }
   };
 };
+
+export const requestSiteReportedPlants = (plantingSiteId: number) => {
+  return async (dispatch: Dispatch, _getState: () => RootState) => {
+    try {
+      const response = await TrackingService.getReportedPlants(plantingSiteId);
+      const { error, site } = response;
+      dispatch(setSiteReportedPlantsAction({ error, site }));
```
Comment (karthikbtf) [inline src/redux/features/tracking/trackingThunks.ts:63]: pass the site id as well

**Review by karthikbtf — COMMENTED** [review]
Looks good in general. Thanks so much for doing this.
Ideally for anything by id, we should book-keep by id (redux state and selectors).
Otherwise there will be race conditions where we load a component and end up showing incorrect data while a request is being dispatched.

**Review by karthikbtf — APPROVED** [review]
Looks good, just that one change if you can 🙏

**Review by karthikbtf — APPROVED** [review]
Sorry for the further suggestions but will be cleaner that way.
Looks great otherwise 🎉

**Comment by charlie-tf** [comment]
> Sorry for the further suggestions but will be cleaner that way. Looks great otherwise 🎉

@karthikbtf thanks for your suggestions in cleaning up this code!

**Review by karthikbtf — APPROVED** [review]
Nice 🎉 
Thanks so much!

**Pull request merged**